include/rmm/mr/device/arena_memory_resource.hpp

```diff
@@ -335,7 +336,8 @@ class arena_memory_resource final : public device_memory_resource {
   void dump_memory_log(size_t bytes)
   {
     logger_->info("**************************************************");
-    logger_->info("Ran out of memory trying to allocate {}.", rmm::detail::bytes{bytes});
+    logger_->info(rmm::detail::formatted_log("Ran out of memory trying to allocate %s.",
+                                             rmm::detail::format_bytes(bytes).c_str()));
```

Comment:
Do you think I should take your "convert_to_c_str" internal function to make it so users don't have to call c_str() everywhere like this?

Reply:
Yes I think that is worthwhile. I wasn't a big fan of my `to_string` template solution since that's an implicit contract that we'd need to teach again, but I think supporting std::string transparently is worthwhile and it's not a surprising behavior.

Reply:
Done.